taskipy/task.py

```diff
@@ -0,0 +1,75 @@
+from typing import Optional
+
+from taskipy.exceptions import InvalidRunnerTypeError, TaskNotFoundError
+from taskipy.pyproject import PyProject
+
+
+class Task:
+    def __init__(self, task_name: str):
+        """
+        Sets up the task by saving it's name and
+        retrieiving the command it references in the
+        pyproject file.
+
+        Args:
+            task_name (str): The name of the task in [tools.taskipy.tasks].
+
+        Raises:
+            TaskNotFoundError: If the task could not be found in the
+                               [tools.taskipy.tasks] section.
+        """
+        self.project = PyProject()
+        self.name = task_name
+
+        try:
```

Comment:
Maybe this should be moved into a property getter, similar to `pre_task` and `post_task`. Move the entire logic of lines 24-27 into `def task(self) -> str` method.

Reply:
Agreed. Perhaps it would be better to leave it named as `command` then for the property, or something else? 

```python
task = Task()
task.task
```

Is a bit odd